logictest: failed to initialize trace dumper: directory to store dumps could not be determined

When logic tests fail in CI, we get the following error:

```

=== RUN   TestLogic/fakedist-vec-off/information_schema
=== PAUSE TestLogic/fakedist-vec-off/information_schema
=== CONT  TestLogic/fakedist-vec-off/information_schema
E210803 00:56:26.842441 5647196 server/server_sql.go:306  [n?] 2252  failed to initialize trace dumper: directory to store dumps could not be determined
E210803 00:56:26.842441 5647196 server/server_sql.go:306  [n?] 2252 +(1) attached stack trace
E210803 00:56:26.842441 5647196 server/server_sql.go:306  [n?] 2252 +  -- stack trace:
E210803 00:56:26.842441 5647196 server/server_sql.go:306  [n?] 2252 +  | github.com/cockroachdb/cockroach/pkg/server/tracedumper.NewTraceDumper
E210803 00:56:26.842441 5647196 server/server_sql.go:306  [n?] 2252 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/server/tracedumper/tracedumper.go:116
E210803 00:56:26.842441 5647196 server/server_sql.go:306  [n?] 2252 +  | github.com/cockroachdb/cockroach/pkg/server.newSQLServer
E210803 00:56:26.842441 5647196 server/server_sql.go:306  [n?] 2252 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/server/server_sql.go:304
E210803 00:56:26.842441 5647196 server/server_sql.go:306  [n?] 2252 +  | github.com/cockroachdb/cockroach/pkg/server.NewServer
E210803 00:56:26.842441 5647196 server/server_sql.go:306  [n?] 2252 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/server/server.go:711
E210803 00:56:26.842441 5647196 server/server_sql.go:306  [n?] 2252 +  | github.com/cockroachdb/cockroach/pkg/server.testServerFactoryImpl.New
E210803 00:56:26.842441 5647196 server/server_sql.go:306  [n?] 2252 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/server/testserver.go:1320
E210803 00:56:26.842441 5647196 server/server_sql.go:306  [n?] 2252 +  | github.com/cockroachdb/cockroach/pkg/testutils/serverutils.NewServer
E210803 00:56:26.842441 5647196 server/server_sql.go:306  [n?] 2252 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/serverutils/test_server_shim.go:289
E210803 00:56:26.842441 5647196 server/server_sql.go:306  [n?] 2252 +  | github.com/cockroachdb/cockroach/pkg/testutils/testcluster.(*TestCluster).AddServer
E210803 00:56:26.842441 5647196 server/server_sql.go:306  [n?] 2252 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/testcluster/testcluster.go:457
E210803 00:56:26.842441 5647196 server/server_sql.go:306  [n?] 2252 +  | github.com/cockroachdb/cockroach/pkg/testutils/testcluster.NewTestCluster
E210803 00:56:26.842441 5647196 server/server_sql.go:306  [n?] 2252 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/testcluster/testcluster.go:271
E210803 00:56:26.842441 5647196 server/server_sql.go:306  [n?] 2252 +  | github.com/cockroachdb/cockroach/pkg/testutils/testcluster.testClusterFactoryImpl.NewTestCluster
E210803 00:56:26.842441 5647196 server/server_sql.go:306  [n?] 2252 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/testcluster/testcluster.go:1548
E210803 00:56:26.842441 5647196 server/server_sql.go:306  [n?] 2252 +  | github.com/cockroachdb/cockroach/pkg/testutils/serverutils.NewTestCluster
E210803 00:56:26.842441 5647196 server/server_sql.go:306  [n?] 2252 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/serverutils/test_cluster_shim.go:229
E210803 00:56:26.842441 5647196 server/server_sql.go:306  [n?] 2252 +  | github.com/cockroachdb/cockroach/pkg/testutils/serverutils.StartNewTestCluster
E210803 00:56:26.842441 5647196 server/server_sql.go:306  [n?] 2252 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/serverutils/test_cluster_shim.go:217
E210803 00:56:26.842441 5647196 server/server_sql.go:306  [n?] 2252 +  | github.com/cockroachdb/cockroach/pkg/sql/logictest.(*logicTest).newCluster
E210803 00:56:26.842441 5647196 server/server_sql.go:306  [n?] 2252 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/logictest/logic.go:1440
E210803 00:56:26.842441 5647196 server/server_sql.go:306  [n?] 2252 +  | github.com/cockroachdb/cockroach/pkg/sql/logictest.(*logicTest).setup
E210803 00:56:26.842441 5647196 server/server_sql.go:306  [n?] 2252 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/logictest/logic.go:1627
E210803 00:56:26.842441 5647196 server/server_sql.go:306  [n?] 2252 +  | github.com/cockroachdb/cockroach/pkg/sql/logictest.RunLogicTestWithDefaultConfig.func1.1
E210803 00:56:26.842441 5647196 server/server_sql.go:306  [n?] 2252 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/logictest/logic.go:3325
E210803 00:56:26.842441 5647196 server/server_sql.go:306  [n?] 2252 +  | testing.tRunner
E210803 00:56:26.842441 5647196 server/server_sql.go:306  [n?] 2252 +  | 	/usr/local/go/src/testing/testing.go:1193
E210803 00:56:26.842441 5647196 server/server_sql.go:306  [n?] 2252 +  | runtime.goexit
E210803 00:56:26.842441 5647196 server/server_sql.go:306  [n?] 2252 +  | 	/usr/local/go/src/runtime/asm_amd64.s:1371
E210803 00:56:26.842441 5647196 server/server_sql.go:306  [n?] 2252 +Wraps: (2) directory to store dumps could not be determined
E210803 00:56:26.842441 5647196 server/server_sql.go:306  [n?] 2252 +Error types: (1) *withstack.withStack (2) *errutil.leafError
        --- FAIL: TestLogic/fakedist-vec-off/information_schema (7.86s)

```

This also shows up on success:
```
 ⚙ otan@otan-mbp  ~/go/src/github.com/cockroachdb/cockroach   datestyle_enabled  make testbaselogic FILES='information_schema'  TESTFLAGS='-rewrite'
Running make with -j16
GOPATH set to /Users/otan/go
find ./pkg -name node_modules -prune -o -type f -name '*.pb.go' -exec rm {} +
find ./pkg -name node_modules -prune -o -type f -name '*.pb.gw.go' -exec rm {} +
mkdir -p lib
buf protoc -Ipkg -I./vendor/github.com/gogo/protobuf -I./vendor/github.com/cockroachdb/errors -I./vendor/go.etcd.io -I./vendor/github.com/prometheus/client_model -I./vendor/github.com/grpc-ecosystem/grpc-gateway/third_party/googleapis --grpc-gateway_out=logtostderr=true,request_context=true:./pkg ./pkg/server/serverpb/admin.proto ./pkg/server/serverpb/status.proto ./pkg/server/serverpb/authentication.proto
set -e; for dir in ./pkg/acceptance/cluster/ ./pkg/blobs/blobspb/ ./pkg/build/ ./pkg/ccl/backupccl/ ./pkg/ccl/baseccl/ ./pkg/ccl/sqlproxyccl/tenant/ ./pkg/ccl/storageccl/engineccl/enginepbccl/ ./pkg/ccl/utilccl/licenseccl/ ./pkg/clusterversion/ ./pkg/config/ ./pkg/config/zonepb/ ./pkg/geo/geoindex/ ./pkg/geo/geopb/ ./pkg/gossip/ ./pkg/jobs/jobspb/ ./pkg/kv/kvnemesis/ ./pkg/kv/kvserver/ ./pkg/kv/kvserver/closedts/ctpb/ ./pkg/kv/kvserver/concurrency/lock/ ./pkg/kv/kvserver/kvserverpb/ ./pkg/kv/kvserver/liveness/livenesspb/ ./pkg/kv/kvserver/protectedts/ptpb/ ./pkg/kv/kvserver/protectedts/ptstorage/ ./pkg/kv/kvserver/readsummary/rspb/ ./pkg/roachpb/ ./pkg/rpc/ ./pkg/server/diagnostics/diagnosticspb/ ./pkg/server/serverpb/ ./pkg/server/status/statuspb/ ./pkg/sql/catalog/descpb/ ./pkg/sql/contentionpb/ ./pkg/sql/execinfrapb/ ./pkg/sql/inverted/ ./pkg/sql/pgwire/pgerror/ ./pkg/sql/schemachanger/scpb/ ./pkg/sql/sessiondatapb/ ./pkg/sql/stats/ ./pkg/sql/types/ ./pkg/startupmigrations/leasemanager/ ./pkg/storage/enginepb/ ./pkg/testutils/grpcutils/ ./pkg/ts/catalog/ ./pkg/ts/tspb/ ./pkg/util/ ./pkg/util/duration/ ./pkg/util/hlc/ ./pkg/util/log/eventpb/ ./pkg/util/log/logpb/ ./pkg/util/metric/ ./pkg/util/optional/ ./pkg/util/protoutil/ ./pkg/util/timeutil/pgdate/ ./pkg/util/tracing/tracingpb/ ./pkg/util/tracing/tracingservicepb/; do \
  buf protoc -Ipkg -I./vendor/github.com/gogo/protobuf -I./vendor/go.etcd.io -I./vendor/github.com/prometheus/client_model -I./vendor/github.com/grpc-ecosystem/grpc-gateway/third_party/googleapis -I./vendor/github.com/cockroachdb/errors --gogoroach_out=Mgoogle/api/annotations.proto=google.golang.org/genproto/googleapis/api/annotations,Mgoogle/protobuf/timestamp.proto=github.com/gogo/protobuf/types,Mgoogle/protobuf/any.proto=github.com/gogo/protobuf/types,Mgoogle/protobuf/duration.proto=github.com/gogo/protobuf/types,plugins=grpc,import_prefix=github.com/cockroachdb/cockroach/pkg/:./pkg $dir/*.proto; \
done
rm -f lib/lib{geos,geos_c}.dylib
cp -L /Users/otan/go/native/x86_64-apple-darwin20.6.0/geos/lib/lib{geos,geos_c}.dylib lib
buf protoc -Ipkg -I./vendor/github.com/gogo/protobuf -I./vendor/github.com/cockroachdb/errors -I./vendor/go.etcd.io -I./vendor/github.com/prometheus/client_model -I./vendor/github.com/grpc-ecosystem/grpc-gateway/third_party/googleapis --grpc-gateway_out=logtostderr=true,request_context=true:./pkg ./pkg/ts/tspb/timeseries.proto
gofmt -s -w ./pkg/server/serverpb/admin.pb.gw.go ./pkg/server/serverpb/status.pb.gw.go ./pkg/server/serverpb/authentication.pb.gw.go ./pkg/ts/tspb/timeseries.pb.gw.go
goimports -w ./pkg/server/serverpb/admin.pb.gw.go ./pkg/server/serverpb/status.pb.gw.go ./pkg/server/serverpb/authentication.pb.gw.go ./pkg/ts/tspb/timeseries.pb.gw.go
touch bin/.gw_protobuf_sources
gofmt -s -w ./pkg/acceptance/cluster/testconfig.pb.go ./pkg/blobs/blobspb/blobs.pb.go ./pkg/build/info.pb.go ./pkg/ccl/backupccl/backup.pb.go ./pkg/ccl/baseccl/encryption_options.pb.go ./pkg/ccl/sqlproxyccl/tenant/directory.pb.go ./pkg/ccl/storageccl/engineccl/enginepbccl/key_registry.pb.go ./pkg/ccl/storageccl/engineccl/enginepbccl/stats.pb.go ./pkg/ccl/utilccl/licenseccl/license.pb.go ./pkg/clusterversion/cluster_version.pb.go ./pkg/config/system.pb.go ./pkg/config/zonepb/zone.pb.go ./pkg/geo/geoindex/config.pb.go ./pkg/geo/geopb/geopb.pb.go ./pkg/gossip/gossip.pb.go ./pkg/jobs/jobspb/jobs.pb.go ./pkg/jobs/jobspb/schedule.pb.go ./pkg/kv/kvnemesis/operations.pb.go ./pkg/kv/kvserver/api.pb.go ./pkg/kv/kvserver/closedts/ctpb/entry.pb.go ./pkg/kv/kvserver/closedts/ctpb/service.pb.go ./pkg/kv/kvserver/concurrency/lock/locking.pb.go ./pkg/kv/kvserver/kvserverpb/lease_status.pb.go ./pkg/kv/kvserver/kvserverpb/log.pb.go ./pkg/kv/kvserver/kvserverpb/proposer_kv.pb.go ./pkg/kv/kvserver/kvserverpb/state.pb.go ./pkg/kv/kvserver/liveness/livenesspb/liveness.pb.go ./pkg/kv/kvserver/protectedts/ptpb/protectedts.pb.go ./pkg/kv/kvserver/protectedts/ptstorage/storage.pb.go ./pkg/kv/kvserver/raft.pb.go ./pkg/kv/kvserver/readsummary/rspb/summary.pb.go ./pkg/kv/kvserver/storage_services.pb.go ./pkg/roachpb/api.pb.go ./pkg/roachpb/app_stats.pb.go ./pkg/roachpb/data.pb.go ./pkg/roachpb/errors.pb.go ./pkg/roachpb/index_usage_stats.pb.go ./pkg/roachpb/internal.pb.go ./pkg/roachpb/internal_raft.pb.go ./pkg/roachpb/io-formats.pb.go ./pkg/roachpb/metadata.pb.go ./pkg/rpc/heartbeat.pb.go ./pkg/server/diagnostics/diagnosticspb/diagnostics.pb.go ./pkg/server/serverpb/admin.pb.go ./pkg/server/serverpb/authentication.pb.go ./pkg/server/serverpb/init.pb.go ./pkg/server/serverpb/migration.pb.go ./pkg/server/serverpb/status.pb.go ./pkg/server/status/statuspb/status.pb.go ./pkg/sql/catalog/descpb/encoded_datum.pb.go ./pkg/sql/catalog/descpb/join_type.pb.go ./pkg/sql/catalog/descpb/locking.pb.go ./pkg/sql/catalog/descpb/privilege.pb.go ./pkg/sql/catalog/descpb/structured.pb.go ./pkg/sql/catalog/descpb/tenant.pb.go ./pkg/sql/contentionpb/contention.pb.go ./pkg/sql/execinfrapb/api.pb.go ./pkg/sql/execinfrapb/component_stats.pb.go ./pkg/sql/execinfrapb/data.pb.go ./pkg/sql/execinfrapb/processors.pb.go ./pkg/sql/execinfrapb/processors_base.pb.go ./pkg/sql/execinfrapb/processors_bulk_io.pb.go ./pkg/sql/execinfrapb/processors_changefeeds.pb.go ./pkg/sql/execinfrapb/processors_sql.pb.go ./pkg/sql/execinfrapb/processors_table_stats.pb.go ./pkg/sql/inverted/span_expression.pb.go ./pkg/sql/pgwire/pgerror/errors.pb.go ./pkg/sql/schemachanger/scpb/scpb.pb.go ./pkg/sql/sessiondatapb/session_data.pb.go ./pkg/sql/stats/histogram.pb.go ./pkg/sql/stats/table_statistic.pb.go ./pkg/sql/types/types.pb.go ./pkg/startupmigrations/leasemanager/lease.pb.go ./pkg/storage/enginepb/engine.pb.go ./pkg/storage/enginepb/file_registry.pb.go ./pkg/storage/enginepb/mvcc.pb.go ./pkg/storage/enginepb/mvcc3.pb.go ./pkg/storage/enginepb/rocksdb.pb.go ./pkg/testutils/grpcutils/testservice.pb.go ./pkg/ts/catalog/chart_catalog.pb.go ./pkg/ts/tspb/timeseries.pb.go ./pkg/util/duration/duration.pb.go ./pkg/util/hlc/legacy_timestamp.pb.go ./pkg/util/hlc/timestamp.pb.go ./pkg/util/log/eventpb/cluster_events.pb.go ./pkg/util/log/eventpb/ddl_events.pb.go ./pkg/util/log/eventpb/events.pb.go ./pkg/util/log/eventpb/health_events.pb.go ./pkg/util/log/eventpb/job_events.pb.go ./pkg/util/log/eventpb/misc_sql_events.pb.go ./pkg/util/log/eventpb/privilege_events.pb.go ./pkg/util/log/eventpb/role_events.pb.go ./pkg/util/log/eventpb/session_events.pb.go ./pkg/util/log/eventpb/sql_audit_events.pb.go ./pkg/util/log/eventpb/zone_events.pb.go ./pkg/util/log/logpb/log.pb.go ./pkg/util/metric/metric.pb.go ./pkg/util/optional/optional.pb.go ./pkg/util/protoutil/clone.pb.go ./pkg/util/timeutil/pgdate/pgdate.pb.go ./pkg/util/tracing/tracingpb/recorded_span.pb.go ./pkg/util/tracing/tracingservicepb/tracing_service.pb.go ./pkg/util/unresolved_addr.pb.go
touch bin/.go_protobuf_sources
go test -c ./pkg/sql/logictest
GOFLAGS= bin/prereqs -bin-name=logictest -test ./pkg/sql/logictest > bin/logictest.d.tmp
mv -f bin/logictest.d.tmp bin/logictest.d
GOFLAGS= go test   -mod=vendor -tags 'make x86_64_apple_darwin20.6.0 crdb_test' -ldflags '-X github.com/cockroachdb/cockroach/pkg/build.typ=development -extldflags ""' -c -o bin/logictest ./pkg/sql/logictest
cd ./pkg/sql/logictest && logictest -test.run "TestLogic//^information_schema$/" -test.timeout 45m -rewrite
initialized metamorphic constant "kv-batch-size" with value 1
initialized metamorphic constant "datum-row-converter-batch-size" with value 1
initialized metamorphic constant "zig-zag-joiner-batch-size" with value 1
initialized metamorphic constant "merge-joiner-groups-buffer" with value 3
initialized metamorphic constant "async-IE-result-channel-buffer-size" with value 15
I210803 02:43:48.540115 1 (gostd) rand.go:92  [-] 1  random seed: -7217600022853504775
E210803 02:43:48.654889 83 server/server_sql.go:306  [n?] 1  failed to initialize trace dumper: directory to store dumps could not be determined
E210803 02:43:48.654889 83 server/server_sql.go:306  [n?] 1 +(1) attached stack trace
E210803 02:43:48.654889 83 server/server_sql.go:306  [n?] 1 +  -- stack trace:
E210803 02:43:48.654889 83 server/server_sql.go:306  [n?] 1 +  | github.com/cockroachdb/cockroach/pkg/server/tracedumper.NewTraceDumper
E210803 02:43:48.654889 83 server/server_sql.go:306  [n?] 1 +  | 	/Users/otan/go/src/github.com/cockroachdb/cockroach/pkg/server/tracedumper/tracedumper.go:116
E210803 02:43:48.654889 83 server/server_sql.go:306  [n?] 1 +  | github.com/cockroachdb/cockroach/pkg/server.newSQLServer
E210803 02:43:48.654889 83 server/server_sql.go:306  [n?] 1 +  | 	/Users/otan/go/src/github.com/cockroachdb/cockroach/pkg/server/server_sql.go:304
E210803 02:43:48.654889 83 server/server_sql.go:306  [n?] 1 +  | github.com/cockroachdb/cockroach/pkg/server.NewServer
E210803 02:43:48.654889 83 server/server_sql.go:306  [n?] 1 +  | 	/Users/otan/go/src/github.com/cockroachdb/cockroach/pkg/server/server.go:711
E210803 02:43:48.654889 83 server/server_sql.go:306  [n?] 1 +  | github.com/cockroachdb/cockroach/pkg/server.testServerFactoryImpl.New
E210803 02:43:48.654889 83 server/server_sql.go:306  [n?] 1 +  | 	/Users/otan/go/src/github.com/cockroachdb/cockroach/pkg/server/testserver.go:1320
E210803 02:43:48.654889 83 server/server_sql.go:306  [n?] 1 +  | github.com/cockroachdb/cockroach/pkg/testutils/serverutils.NewServer
E210803 02:43:48.654889 83 server/server_sql.go:306  [n?] 1 +  | 	/Users/otan/go/src/github.com/cockroachdb/cockroach/pkg/testutils/serverutils/test_server_shim.go:289
E210803 02:43:48.654889 83 server/server_sql.go:306  [n?] 1 +  | github.com/cockroachdb/cockroach/pkg/testutils/testcluster.(*TestCluster).AddServer
E210803 02:43:48.654889 83 server/server_sql.go:306  [n?] 1 +  | 	/Users/otan/go/src/github.com/cockroachdb/cockroach/pkg/testutils/testcluster/testcluster.go:457
E210803 02:43:48.654889 83 server/server_sql.go:306  [n?] 1 +  | github.com/cockroachdb/cockroach/pkg/testutils/testcluster.NewTestCluster
E210803 02:43:48.654889 83 server/server_sql.go:306  [n?] 1 +  | 	/Users/otan/go/src/github.com/cockroachdb/cockroach/pkg/testutils/testcluster/testcluster.go:271
E210803 02:43:48.654889 83 server/server_sql.go:306  [n?] 1 +  | github.com/cockroachdb/cockroach/pkg/testutils/testcluster.testClusterFactoryImpl.NewTestCluster
E210803 02:43:48.654889 83 server/server_sql.go:306  [n?] 1 +  | 	/Users/otan/go/src/github.com/cockroachdb/cockroach/pkg/testutils/testcluster/testcluster.go:1548
E210803 02:43:48.654889 83 server/server_sql.go:306  [n?] 1 +  | github.com/cockroachdb/cockroach/pkg/testutils/serverutils.NewTestCluster
E210803 02:43:48.654889 83 server/server_sql.go:306  [n?] 1 +  | 	/Users/otan/go/src/github.com/cockroachdb/cockroach/pkg/testutils/serverutils/test_cluster_shim.go:229
E210803 02:43:48.654889 83 server/server_sql.go:306  [n?] 1 +  | github.com/cockroachdb/cockroach/pkg/testutils/serverutils.StartNewTestCluster
E210803 02:43:48.654889 83 server/server_sql.go:306  [n?] 1 +  | 	/Users/otan/go/src/github.com/cockroachdb/cockroach/pkg/testutils/serverutils/test_cluster_shim.go:217
E210803 02:43:48.654889 83 server/server_sql.go:306  [n?] 1 +  | github.com/cockroachdb/cockroach/pkg/sql/logictest.(*logicTest).newCluster
E210803 02:43:48.654889 83 server/server_sql.go:306  [n?] 1 +  | 	/Users/otan/go/src/github.com/cockroachdb/cockroach/pkg/sql/logictest/logic.go:1440
E210803 02:43:48.654889 83 server/server_sql.go:306  [n?] 1 +  | github.com/cockroachdb/cockroach/pkg/sql/logictest.(*logicTest).setup
E210803 02:43:48.654889 83 server/server_sql.go:306  [n?] 1 +  | 	/Users/otan/go/src/github.com/cockroachdb/cockroach/pkg/sql/logictest/logic.go:1627
E210803 02:43:48.654889 83 server/server_sql.go:306  [n?] 1 +  | github.com/cockroachdb/cockroach/pkg/sql/logictest.RunLogicTestWithDefaultConfig.func1.1
E210803 02:43:48.654889 83 server/server_sql.go:306  [n?] 1 +  | 	/Users/otan/go/src/github.com/cockroachdb/cockroach/pkg/sql/logictest/logic.go:3325
E210803 02:43:48.654889 83 server/server_sql.go:306  [n?] 1 +  | testing.tRunner
E210803 02:43:48.654889 83 server/server_sql.go:306  [n?] 1 +  | 	/usr/local/opt/go/libexec/src/testing/testing.go:1193
E210803 02:43:48.654889 83 server/server_sql.go:306  [n?] 1 +  | runtime.goexit
E210803 02:43:48.654889 83 server/server_sql.go:306  [n?] 1 +  | 	/usr/local/opt/go/libexec/src/runtime/asm_amd64.s:1371
E210803 02:43:48.654889 83 server/server_sql.go:306  [n?] 1 +Wraps: (2) directory to store dumps could not be determined
E210803 02:43:48.654889 83 server/server_sql.go:306  [n?] 1 +Error types: (1) *withstack.withStack (2) *errutil.leafError
E210803 02:43:52.051717 448 sql/stats/automatic_stats.go:298  [-] 2  failed to refresh stats: node unavailable; try another peer
PASS
```

See also: https://teamcity.cockroachdb.com/viewLog.html?buildId=3254204&buildTypeId=Cockroach_UnitTests

--- a/pkg/server/server_sql.go
+++ b/pkg/server/server_sql.go
@@ -301,10 +301,8 @@ func newSQLServer(ctx context.Context, cfg sqlServerArgs) (*SQLServer, error) {
 		if cfg.TestingKnobs.JobsTestingKnobs != nil {
 			jobsKnobs = cfg.TestingKnobs.JobsTestingKnobs.(*jobs.TestingKnobs)
 		}
-		td, err := tracedumper.NewTraceDumper(ctx, cfg.InflightTraceDirName, cfg.Settings)
-		if err != nil {
-			log.Errorf(ctx, "failed to initialize trace dumper: %+v", err)
-		}
+
+		td := tracedumper.NewTraceDumper(ctx, cfg.InflightTraceDirName, cfg.Settings)
 		*jobRegistry = *jobs.MakeRegistry(
 			cfg.AmbientCtx,
 			cfg.stopper,

--- a/pkg/server/testserver.go
+++ b/pkg/server/testserver.go
@@ -256,6 +256,9 @@ func makeTestConfigFromParams(params base.TestServerArgs) Config {
 			if cfg.GoroutineDumpDirName == "" {
 				cfg.GoroutineDumpDirName = filepath.Join(storeSpec.Path, "logs", base.GoroutineDumpDir)
 			}
+			if cfg.InflightTraceDirName == "" {
+				cfg.InflightTraceDirName = filepath.Join(storeSpec.Path, "logs", base.InflightTraceDir)
+			}
 		}
 	}
 	cfg.Stores = base.StoreSpecList{Specs: params.StoreSpecs}

--- a/pkg/server/tracedumper/tracedumper.go
+++ b/pkg/server/tracedumper/tracedumper.go
@@ -111,9 +111,9 @@ func (t *TraceDumper) Dump(
 // NewTraceDumper returns a TraceDumper.
 //
 // dir is the directory in which dumps are stored.
-func NewTraceDumper(ctx context.Context, dir string, st *cluster.Settings) (*TraceDumper, error) {
+func NewTraceDumper(ctx context.Context, dir string, st *cluster.Settings) *TraceDumper {
 	if dir == "" {
-		return nil, errors.New("directory to store dumps could not be determined")
+		return nil
 	}
 
 	log.Infof(ctx, "writing job trace dumps to %s", dir)
@@ -122,5 +122,5 @@ func NewTraceDumper(ctx context.Context, dir string, st *cluster.Settings) (*Tra
 		currentTime: timeutil.Now,
 		store:       dumpstore.NewStore(dir, totalDumpSizeLimit, st),
 	}
-	return td, nil
+	return td
 }

